### PR TITLE
Fix simple policy implementation for outgoing transactions

### PR DESCRIPTION
**Pull request by jseagrave21**
**What current issue(s) does this address, or what feature is it adding?**
Addresses #960; specifically, https://github.com/CityOfZion/neo-python/issues/960#issuecomment-512248474

**How did you solve this problem?**
implement SimplePolicy check when relaying any tx
- results in a cancelled tx and a required fee recommendation if SimplePolicy requirements are not met
- addresses https://github.com/CityOfZion/neo-python/issues/960#issuecomment-512248474 by checking each tx after it is signed

**How did you make sure your solution works?**
`make test` with unittests

**Are there any special changes in the code that we should be aware of?**
Replaced `TXFeeError` with `TXError` to incorporate a check for violating the tx maximum size

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [x] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

**Comment by ixje** [comment]
note to self; gave slack feedback regarding late notification if funds are insufficient

File: neo/Core/TX/Transaction.py [hunk 1]
```diff
@@ -775,6 +775,6 @@ def __init__(self, *args, **kwargs):
         self.Type = TransactionType.ContractTransaction
 
 
-class TXFeeError(Exception):
-    """Provide user-friendly feedback for transaction fee errors."""
+class TXError(Exception):
+    """Provide user-friendly feedback for transaction errors."""
```
Comment (jseagrave21) [inline neo/Core/TX/Transaction.py]: It seems like `TXError` is now obsolete. Is there any reason to keep it?

**Comment by ixje** [comment]
I had another look at the code because it just didn't feel right to have `relay()` be responsible for policy validation. One of the issues in resolving this in the best way possible is that certain code parts do too much e.g. https://github.com/CityOfZion/neo-python/blob/fe90f62e123d720d4281c79af0598d9df9e776fb/neo/Prompt/Commands/Invoke.py#L48

does not just invoke a contract, it also sign's the TX and this is a problem for reporting the required fees prior to asking if the TX should be send to the network. Instead of resolving the latter now I want to keep that in mind for the NEO3 build.

I moved the policy validation into its own function and out of the `relay()` code because those 2 should not be tied together. In doing so I could also simplify some of the test logic.

File: neo/Core/TX/Transaction.py (hunk 1 above)
Comment (ixje) [reply neo/Core/TX/Transaction.py]: I guess not.

**Comment by ixje** [comment]
🎉

**Pull request merged**